google/cloud/ndb/model.py

```diff
@@ -2527,9 +2527,20 @@ def _to_datastore(self, entity, data, prefix="", repeated=False):
             if isinstance(value, _CompressedValue):
                 value = value.z_val
                 data[self._name] = value
-            if value and not value.startswith(_ZLIB_COMPRESSION_MARKER):
-                value = zlib.compress(value)
+
+            if self._repeated:
+                compressed_value = []
+                for rval in value:
+                    if rval and not rval.startswith(_ZLIB_COMPRESSION_MARKER):
+                        rval = zlib.compress(rval)
+                    compressed_value.append(rval)
+                value = compressed_value
                 data[self._name] = value
+            if not self._repeated:
```

Comment:
Why not just `else:`?

Reply:
You are right. I was too focused on the distinction between the two. However, it doesn't hurt, so I guess I'll leave it as is.